### PR TITLE
Test Classes Refactoring for setters of _textIndexColumns and ColumnProperties (IndexLoadingConfig) 

### PR DESCRIPTION
**Pull request by deepthi912**
Follow up for https://github.com/apache/pinot/issues/14106

Below test classes are modified:
1) NativeAndLuceneComparisonTest
2) NoDictionaryCompressionQueriesTest
3) TextSearchQueriesTest
4) FastFilteredCountTest
5) BenchmarkNativeVsLuceneTextIndex

### CODE REVIEW

File: pinot-core/src/test/java/org/apache/pinot/queries/NoDictionaryCompressionQueriesTest.java [hunk 1]
```diff
@@ -96,6 +94,8 @@ public class NoDictionaryCompressionQueriesTest extends BaseQueriesTest {
   private IndexSegment _indexSegment;
   private List<IndexSegment> _indexSegments;
   private List<GenericRow> _rows;
+  private TableConfig _tableConfig;
+  Schema _schema;
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/queries/NoDictionaryCompressionQueriesTest.java:98]: (minor)
```suggestion
  private Schema _schema;
```

File: pinot-core/src/test/java/org/apache/pinot/queries/FastFilteredCountTest.java [hunk 2]
```diff
@@ -123,11 +122,16 @@ public void setUp()
...
     driver.build();
 
-    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
-    indexLoadingConfig.setInvertedIndexColumns(new HashSet<>(Arrays.asList(CLASSIFICATION_COLUMN, SORTED_COLUMN)));
-    indexLoadingConfig.setTextIndexColumns(Collections.singleton(TEXT_COLUMN));
-    indexLoadingConfig.setJsonIndexColumns(Collections.singleton(JSON_COLUMN));
-    indexLoadingConfig.setRangeIndexColumns(Collections.singleton(INT_RANGE_COLUMN));
+    List<FieldConfig> fieldConfigs = new ArrayList<>();
+    fieldConfigs.add(
+        new FieldConfig(TEXT_COLUMN, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT, null, null));
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/queries/FastFilteredCountTest.java:127]: (minor)
```suggestion
    List<FieldConfig> fieldConfigs = List.of(new FieldConfig(TEXT_COLUMN, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT, null, null)));
```

File: pinot-core/src/test/java/org/apache/pinot/queries/FastFilteredCountTest.java [hunk 3]
```diff
@@ -123,11 +122,16 @@ public void setUp()
...
+
+    TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName(RAW_TABLE_NAME)
+        .setInvertedIndexColumns(Arrays.asList(CLASSIFICATION_COLUMN, SORTED_COLUMN))
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/queries/FastFilteredCountTest.java:130]: (minor) Change all `Arrays.asList()` to `List.of()` which has slightly better performance

File: pinot-core/src/test/java/org/apache/pinot/queries/NativeAndLuceneComparisonTest.java [hunk 4]
```diff
@@ -219,38 +217,41 @@ private void buildNativeTextIndexSegment()
...
-    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
-    Set<String> textIndexCols = new HashSet<>();
-    textIndexCols.add(QUOTES_COL_LUCENE);
-    textIndexCols.add(QUOTES_COL_LUCENE_MV);
-    indexLoadingConfig.setTextIndexColumns(textIndexCols);
-    Set<String> invertedIndexCols = new HashSet<>();
-    invertedIndexCols.add(QUOTES_COL_LUCENE);
-    invertedIndexCols.add(QUOTES_COL_LUCENE_MV);
-    indexLoadingConfig.setInvertedIndexColumns(invertedIndexCols);
+    List<FieldConfig> fieldConfigs = new ArrayList<>();
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/queries/NativeAndLuceneComparisonTest.java:220]: (minor) Similarly here, we may use `List.of()` and `Map.of()` for concise. Same for other places

File: pinot-core/src/test/java/org/apache/pinot/queries/TextSearchQueriesTest.java [hunk 5]
```diff
@@ -138,27 +146,41 @@ protected List<IndexSegment> getIndexSegments() {
...
-    columnProperties.put(SKILLS_TEXT_COL_DICT_NAME, props);
-    indexLoadingConfig.setColumnProperties(columnProperties);
+    fieldConfigs.add(
+        new FieldConfig(SKILLS_TEXT_COL_DICT_NAME, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT,
+            null, props));
+    fieldConfigs.add(
+        new FieldConfig(SKILLS_TEXT_MV_COL_DICT_NAME, FieldConfig.EncodingType.DICTIONARY, FieldConfig.IndexType.TEXT,
+            null, null));
+
+    _tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName(TABLE_NAME)
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/queries/TextSearchQueriesTest.java:180]: Seems we are not reusing table config here. Do we need to construct a new one from scratch?

**Review by Jackie-Jiang — APPROVED** [review]
LGTM in general

File: pinot-core/src/test/java/org/apache/pinot/queries/TextSearchQueriesTest.java (hunk 5 above)
Comment (deepthi912) [reply pinot-core/src/test/java/org/apache/pinot/queries/TextSearchQueriesTest.java:180]: In buildSegment() we created one which doesn't include any column properties for field configs to be used in SegmentGeneratorConfig, so tried to create a new one for this with those details included.

**Pull request merged**